tests/e2e_tests/bp_map_e2e.py

```diff
@@ -0,0 +1,327 @@
+
+    generated_bp_map_file = os.path.join(bp_map_outputdir, "dark_mock_bad_pixel_map.fits")
+
+    # Load the generated bad pixel map image and master dark reference data
+    generated_bp_map_img = data.Image(generated_bp_map_file)
+
+    with fits.open(master_dark_ref) as hdulist:
+        dark_ref_dat = hdulist[1].data
+
+    with fits.open(dark_current_path) as hdulist:
+        naxis1 = hdulist[0].header['NAXIS1']  # X-axis size
+        naxis2 = hdulist[0].header['NAXIS2']  # Y-axis size
+
+    with fits.open(bp_ref_path) as hdulist:
+        bp_ref_dat = hdulist[0].data
+
+        diff = generated_bp_map_img.data - bp_ref_dat.data
+
+        # Check for differences between the generated and reference bad pixel maps
+        assert np.all(np.abs(diff) < 1e-5)
+
+        # Identify and print any bad pixels found
+        bad_pixels = generated_bp_map_img.data[generated_bp_map_img.data > 0]
+        if bad_pixels.size > 0:
+            print(f"Bad pixel values identified: {bad_pixels}")
+        else:
+            print("No bad pixels identified")
+
+        # Plot noise maps, master dark, and bad pixel maps
+        fig, axes = plt.subplots(2, 3, figsize=(15, 10))
+        axes = axes.flatten()
+
+        # Plot each subplot with relevant data and titles
+        im1 = axes[0].imshow(fpn_dat, vmin=-240, vmax=240, cmap="gray")
+        axes[0].set_title("FPN noise map")
+        axes[0].set_xlim([0, naxis1])
+        axes[0].set_ylim([0, naxis2])
+        fig.colorbar(im1, ax=axes[0])
+
+        im2 = axes[1].imshow(cic_dat, vmin=0, vmax=0.02, cmap="gray")
+        axes[1].set_title("CIC noise map")
+        axes[1].set_xlim([0, naxis1])
+        axes[1].set_ylim([0, naxis2])
+        fig.colorbar(im2, ax=axes[1])
+
+        im3 = axes[2].imshow(dark_current_dat, vmin=0, vmax=0.003, cmap="gray")
+        axes[2].set_title("Dark current noise map")
+        axes[2].set_xlim([0, naxis1])
+        axes[2].set_ylim([0, naxis2])
+        fig.colorbar(im3, ax=axes[2])
+
+        im4 = axes[3].imshow(dark_ref_dat, vmin=-240, vmax=240, cmap="gray")
+        axes[3].set_title("DRP-produced master dark")
+        axes[3].set_xlim([0, naxis1])
+        axes[3].set_ylim([0, naxis2])
+        fig.colorbar(im4, ax=axes[3])
+
+        im5 = axes[4].imshow(generated_bp_map_img.data, vmin=0, vmax=8, cmap="gray")
+        axes[4].set_title("DRP-produced BP map")
+        axes[4].set_xlim([0, naxis1])
+        axes[4].set_ylim([0, naxis2])
+        fig.colorbar(im5, ax=axes[4])
+
+        im6 = axes[5].imshow(bp_ref_dat, vmin=0, vmax=8, cmap="gray")
+        axes[5].set_title("Reference BP map")
+        axes[5].set_xlim([0, naxis1])
+        axes[5].set_ylim([0, naxis2])
+        fig.colorbar(im6, ax=axes[5])
+
+        plt.tight_layout()
+
+        # Save the figure to a file
+        output_path = os.path.join(bp_map_outputdir, "bp_map_master_dark_test.png")
+        plt.savefig(output_path)
+
+def bp_map_simulated_dark_e2e(tvacdata_path, e2eoutput_path):
```

Comment:
In addition to preprending "test_", we also need the `@pytest.mark.e2e` decorator.

Reply:
done